src/containers/Classroom.jsx

```diff
@@ -10,19 +10,21 @@ import { deleteClassroom, deleteStudent } from '../actions/teacher';
 export default class Classroom extends Component {
 
   render() {
-    const { classrooms, params, actions } = this.props;
+    const { classrooms, assignments, params, actions } = this.props;
     const members = classrooms.members;
     const classroom = classrooms.data.find(classroom => classroom.id === params.classroomId);
     const classroomId = (classroom && classroom.id) ? classroom.id : undefined;
     const boundDeleteClassroom = actions.deleteClassroom.bind(this, classroomId);
     const students = classroom ? classroom.relationships.students.data : undefined;
     const studentIds = students ? students.map(student => student.id) : undefined;
     const boundDeleteStudent = actions.deleteStudent.bind(this);
+    const classroomAssignments = assignments;
```

Comment:
Ah, good catch, that code should have read...

```
const classroomAssignments = assignments;
//TODO: Select the correct subset of assignments for this class.
```

The `assignments` value lists all assignments, and we only want the ones related to _this classroom._

So it should be something like...

```
assignments = [
  { name: "lots of lions", id="1" classroom_id="337" }
  { name: "bunch of baboons", id="2" classroom_id="911" }
  { name: "crapload of cougars", id="3" classroom_id="337" }
];

classroomID = 337;
classroomAssignments = [
  { name: "lots of lions", id="1" classroom_id="337" }
  { name: "crapload of cougars", id="3" classroom_id="337" }
];

```

I believe @marten already made some changes to the EduAPI that marks each assignment with its respective Class ID, but I still need to check if it's live/deployed.

(Note: there is an option to select Assignments by Class ID, but that will involve extra calls to the API - once for every time a new Class is selected. This is sub-optimal, but a possible Plan B.)